### PR TITLE
Add metadata to timegraph deserializer

### PR DESCRIPTION
**Pull request by bhufmann**
This PR adds support for metadata in the timegraph deserializer. It includes also unit tests to verify this change.

Signed-off-by: Rodrigo Pinto <rodrigo.pinto@calian.ca>
Signed-off-by: Bernd Hufmann <bernd.hufmann@ericsson.com>

### CODE REVIEW

**Comment by bhufmann** [comment]
This PR is based on PR #62. It includes a modified version of #62 plus unit tests.

File: tsp-typescript-client/src/models/entry.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ export const Entry = createNormalizer<Entry>({
     style: {
         values: undefined,
     },
+    metadata: undefined
```
Comment (marco-miller) [inline tsp-typescript-client/src/models/entry.ts:10]: The style of this file would expect a trailing comma even here too.

File: tsp-typescript-client/src/protocol/tsp-client.test.ts [hunk 2]
```diff
@@ -279,6 +279,17 @@ describe('TspClient Deserialization', () => {
       expect(typeof entry.end).toEqual('bigint');
       expect(typeof entry.id).toEqual('number');
       expect(typeof entry.start).toEqual('bigint');
+      const metadata = entry.metadata;
+      expect(metadata).toBeDefined();
+      const pids = metadata?.pid;
+      expect(pids).toBeDefined();
+      expect(pids).toHaveLength(1);
+      const tids = metadata?.pid;
```
Comment (marco-miller) [inline tsp-typescript-client/src/protocol/tsp-client.test.ts:287]: .tids, it seems.

File: tsp-typescript-client/src/protocol/tsp-client.test.ts [hunk 3]
```diff
@@ -279,6 +279,17 @@ describe('TspClient Deserialization', () => {
       expect(typeof entry.end).toEqual('bigint');
       expect(typeof entry.id).toEqual('number');
       expect(typeof entry.start).toEqual('bigint');
+      const metadata = entry.metadata;
+      expect(metadata).toBeDefined();
+      const pids = metadata?.pid;
```
Comment (marco-miller) [inline tsp-typescript-client/src/protocol/tsp-client.test.ts:284]: .pids, rather?

File: tsp-typescript-client/src/protocol/tsp-client.test.ts [hunk 4]
```diff
@@ -279,6 +279,17 @@ describe('TspClient Deserialization', () => {
...
+      expect(pids).toBeDefined();
+      expect(pids).toHaveLength(1);
+      const tids = metadata?.pid;
+      expect(tids).toBeDefined();
+      expect(tids).toHaveLength(1);
+      const exec = metadata?.pid;
```
Comment (marco-miller) [inline tsp-typescript-client/src/protocol/tsp-client.test.ts:290]: .exec instead?

File: tsp-typescript-client/src/protocol/tsp-client.test.ts [hunk 5]
```diff
@@ -279,6 +279,17 @@ describe('TspClient Deserialization', () => {
       expect(typeof entry.end).toEqual('bigint');
       expect(typeof entry.id).toEqual('number');
       expect(typeof entry.start).toEqual('bigint');
+      const metadata = entry.metadata;
+      expect(metadata).toBeDefined();
+      const pids = metadata?.pid;
+      expect(pids).toBeDefined();
```
Comment (marco-miller) [inline tsp-typescript-client/src/protocol/tsp-client.test.ts:285]: If using .pids instead above, this check fails as that gets undefined.

Maybe passing this somehow will lead to failing more checks below, if more get undefined, or if any actual length isn't 1.

**Review by marco-miller — CHANGES_REQUESTED** [review]

File: tsp-typescript-client/src/models/timegraph.ts [hunk 6]
```diff
@@ -8,6 +8,7 @@ export const TimeGraphEntry = createNormalizer<TimeGraphEntry>({
     parentId: assertNumber,
     start: BigInt,
     style: OutputElementStyle,
+    metadata: undefined
```
[the remark this answers is not in the document]
Comment (bhufmann) [reply tsp-typescript-client/src/models/timegraph.ts:11]: Done

File: tsp-typescript-client/src/protocol/tsp-client.test.ts (hunk 2 above)
Comment (bhufmann) [reply tsp-typescript-client/src/protocol/tsp-client.test.ts:287]: copy-paste error... I'll fix it

File: tsp-typescript-client/src/protocol/tsp-client.test.ts (hunk 3 above)
Comment (bhufmann) [reply tsp-typescript-client/src/protocol/tsp-client.test.ts:284]: Done

File: tsp-typescript-client/src/protocol/tsp-client.test.ts (hunk 4 above)
Comment (bhufmann) [reply tsp-typescript-client/src/protocol/tsp-client.test.ts:290]: Done

File: tsp-typescript-client/src/protocol/tsp-client.test.ts (hunk 5 above)
Comment (bhufmann) [reply tsp-typescript-client/src/protocol/tsp-client.test.ts:285]: In the test file the key is define as `pid`.

File: tsp-typescript-client/src/models/entry.ts (hunk 1 above)
Comment (bhufmann) [reply tsp-typescript-client/src/models/entry.ts:10]: Done

File: tsp-typescript-client/src/models/timegraph.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (marco-miller) [reply tsp-typescript-client/src/models/timegraph.ts:11]: Pushed?

File: tsp-typescript-client/src/protocol/tsp-client.test.ts (hunk 4 above)
Comment (marco-miller) [reply tsp-typescript-client/src/protocol/tsp-client.test.ts]: You changed it to `exec_name`, rather, which LGTM.

**Review by marco-miller — APPROVED** [review]

**Pull request merged**